types/relay-runtime/index.d.ts

```diff
@@ -46,880 +46,157 @@ import {
     NormalizationHandle,
 } from './lib/util/NormalizationNode';
 
-// ./mutations/RelayDeclarativeMutationConfig
-interface RangeAddConfig {
-    type: 'RANGE_ADD';
-    parentName?: string;
-    parentID?: string;
-    connectionInfo?: ReadonlyArray<{
-        key: string;
-        filters?: Variables;
-        rangeBehavior: string;
-    }>;
-    connectionName?: string;
-    edgeName: string;
-    rangeBehaviors?: RangeBehaviors;
-}
-
-interface RangeDeleteConfig {
-    type: 'RANGE_DELETE';
-    parentName?: string;
-    parentID?: string;
-    connectionKeys?: ReadonlyArray<{
-        key: string;
-        filters?: Variables;
-    }>;
-    connectionName?: string;
-    deletedIDFieldName: string | ReadonlyArray<string>;
-    pathToConnection: ReadonlyArray<string>;
-}
-
-interface NodeDeleteConfig {
-    type: 'NODE_DELETE';
-    parentName?: string;
-    parentID?: string;
-    connectionName?: string;
-    deletedIDFieldName: string;
-}
-
-// Unused in Relay Modern
-interface LegacyFieldsChangeConfig {
-    type: 'FIELDS_CHANGE';
-    fieldIDs: { [fieldName: string]: DataID | ReadonlyArray<DataID> };
-}
-
-// Unused in Relay Modern
-interface LegacyRequiredChildrenConfig {
-    type: 'REQUIRED_CHILDREN';
-    children: ReadonlyArray<unknown>;
-}
-export type DeclarativeMutationConfig =
-    | RangeAddConfig
-    | RangeDeleteConfig
-    | NodeDeleteConfig
-    | LegacyFieldsChangeConfig
-    | LegacyRequiredChildrenConfig;
-export type MutationType = 'RANGE_ADD' | 'RANGE_DELETE' | 'NODE_DELETE' | 'FIELDS_CHANGE' | 'REQUIRED_CHILDREN';
-export type RangeOperation = 'append' | 'ignore' | 'prepend' | 'refetch' | 'remove';
-export type RangeBehaviors =
-    | ((connectionArgs: { [name: string]: unknown }) => RangeOperation)
-    | {
-          [key: string]: RangeOperation;
-      };
-
-// ./mutations/applyRelayModernOptimisticMutation
-export interface OptimisticMutationConfig {
-    configs?: ReadonlyArray<DeclarativeMutationConfig> | null;
-    mutation: GraphQLTaggedNode;
-    variables: Variables;
-    optimisticUpdater?: SelectorStoreUpdater | null;
-    optimisticResponse?: object;
-}
-
-// ./mutations/commitRelayModernMutation
-export interface MutationConfig<TOperation extends OperationType> {
-    configs?: ReadonlyArray<DeclarativeMutationConfig>;
-    mutation: GraphQLTaggedNode;
-    variables: TOperation['variables'];
-    uploadables?: UploadableMap;
-    onCompleted?:
-        | ((response: TOperation['response'], errors: ReadonlyArray<PayloadError> | null | undefined) => void)
-        | null;
-    onError?: ((error: Error) => void) | null;
-    optimisticUpdater?: SelectorStoreUpdater | null;
-    optimisticResponse?: TOperation['response'] | null;
-    updater?: SelectorStoreUpdater<TOperation['response']> | null;
-}
+// ./lib/mutations/RelayDeclarativeMutationConfig
```

Comment:
There’s no more need to have these comments, as that’s self-documenting now.

Reply:
Removed. Follow the exports of source code of [relay-runtime](https://github.com/facebook/relay/blob/master/packages/relay-runtime/index.js)